### ISSUE TITLE
Platform xView dataset makes most targets tiny and caps validation recall at 23.6%

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/ultralytics#25992]
## Summary

The [Ultralytics Platform xView dataset](https://platform.ultralytics.com/ultralytics/datasets/xview) currently stores the
original high-resolution source scenes as individual training and validation images. This makes the documented/default
`imgsz=640` training setup unsuitable for xView's dense, very small objects, and `max_det=300` imposes a severe validation
ceiling.

This is a dataset-representation problem rather than a TaskAlignedAssigner problem. The improved OOM recovery in #25990
makes these dense batches more tractable, but it cannot restore spatial information discarded by resizing or recover objects
beyond `max_det`.

## Measured Platform dataset

Dataset metadata:

```text
URL: https://platform.ultralytics.com/ultralytics/datasets/xview
hash: c30b33cb
train: 760 images, 537,009 objects
val: 86 images, 64,686 objects
median source image: approximately 3,200 x 2,800 px
```

Projected minimum box dimensions after letterboxing each source image to `imgsz=640`:

| Split | Below 1 px | Below 2 px | Below 4 px | Below 8 px |
| --- | ---: | ---: | ---: | ---: |
| Train | 0.9% | 24.3% | 51.2% | 79.0% |
| Val | 0.7% | 25.0% | 53.1% | 78.4% |

The validation median minimum box dimension is only 3.56 px. More than half of all validation objects therefore have a
minimum dimension below four input pixels.

The validation density also conflicts with `max_det=300`:

```text
validation images: 86
images with more than 300 labels: 40
maximum labels in one image: 5,609
total labels: 64,686
maximum possible true positives with max_det=300: 15,285
theoretical aggregate recall ceiling: 15,285 / 64,686 = 23.6%
```

`max_det` is applied per image by the detection head and validator. Across the current label distribution, even perfect
ranking and localization cannot exceed this aggregate true-positive ceiling.

## Reproduction

After downloading the Platform dataset, calculate projected box sizes from its YOLO labels with
`scale = 640 / max(image_width, image_height)`. Calculate the `max_det` ceiling as:

```python
maximum_true_positives = sum(min(number_of_labels, 300) for each_validation_image)
recall_ceiling = maximum_true_positives / total_validation_labels
```

This was reproduced on Ultralytics 8.4.134 at main commit `1500fddde023d5ddac7029ed1018d2bf4c3cb75d` using the fully
materialized Platform dataset above.

## Impact

- A large fraction of targets become effectively sub-pixel or only a few pixels wide.
- Validation recall and therefore mAP are constrained by `max_det`, independently of model quality.
- Extremely dense source images create unusually large assigner workloads.
- In multi-dataset benchmarks such as UL37, the xView fitness reflects this representation and output cap more than useful
  hyperparameter quality.
- Raising global `imgsz` or `max_det` would make every other dataset substantially more expensive and still would not fully
  address xView's scale mismatch.

## Proposed owner-level fix

Publish xView on Platform as a deterministic native-resolution tiled dataset rather than resizing complete source scenes:

1. Split by original source image before tiling to prevent spatial train/validation leakage.
2. Generate native-resolution `640 x 640` tiles, padding edge tiles.
3. Clip boxes at tile boundaries and discard only degenerate fragments.
4. Materialize a deterministic 1,000-tile training split with broad source/class coverage for `fraction=[1000, 1.0, 0]`.
5. Tile the complete validation split and retain empty validation tiles so false positives remain measurable.
6. Publish a new dataset version/hash or slug so existing full-image caches cannot be mistaken for the corrected dataset.

Tiling preserves the native 5-10 px vehicle signal, distributes dense labels across manageable images, and makes
`max_det=300` reasonable without changing global training arguments. It also follows the preprocessing strategy used by the
original xView baseline, which trained on small chips rather than globally downscaled scenes.

## Acceptance criteria

- No train/validation leakage between tiles from the same source image.
- The standard `imgsz=640`, `max_det=300` path no longer creates a structural recall ceiling.
- Dataset metadata documents the tiling policy and retains source provenance/license.
- Platform download/materialization produces a new cache identity.
- A short train/validation smoke test confirms ordinary GPU assignment without the current extreme per-image label counts.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for the detailed analysis and proposed fix. The current full-resolution xView representation creates a severe small-object scale mismatch at `imgsz=640`, while dense validation images make `max_det=300` impose a structural recall ceiling. The deterministic native-resolution tiling approach, source-level split protection, retained empty validation tiles, provenance, and new dataset cache identity are clear acceptance criteria for evaluation by the Platform team.

This is an automated response, and an Ultralytics engineer will assist soon with triage and next steps.

**Comment by glenn-jocher** [comment]
Fixed in #25993
